js/src/main/scala/org/widok/validation/Validator.scala

```diff
@@ -0,0 +1,28 @@
+package org.widok.validation
+
+import org.widok.{Dict, ReadStateChannel}
+
+case class Validator(validationSources: Tuple2[ReadStateChannel[_], Seq[FieldValidation[_]]]*) {
+
+  val validations = Dict[ReadStateChannel[_], Seq[String]]()
+
+  val valid = validations.forall(_.isEmpty).cache(true)
+  val errors = validations.filter(_.nonEmpty).buffer
+
+  validationSources.foreach {
+    case (channel, fieldValidators) =>
+      // don't validate initial values (tail) and only validate when the value is actually updated (distinct)
+      channel.distinct.tail.attach(input => validateValue(channel, fieldValidators, input))
+  }
+
+  private def validateValue(channel: ReadStateChannel[_], fieldValidators: Seq[FieldValidation[_]], input: Any) = {
```

Comment:
The equal sign is gone.

I've been trying (unsuccessfully) to use a type parameter. The problem is that the validationSources needs to have en "open type" since the channels type is unknown. That's the reason why I asked if the channels for input-fields could be of any other types then strings (date, numbers, ..). But even if one could assume that all input field channels are Strings (to remove the type all together), there is the case with combined fields (password check) etc.

So, i don't see how to add a type to validateValue when it's called from the validationSources.foreach for example?